components/tools/OmeroPy/src/omero/plugins/web.py

```diff
@@ -210,20 +221,40 @@ def _set_apache_wsgi(self, d, settings):
         d["OMEROPYTHONROOT"] = self._get_python_dir()
         d["OMEROFALLBACKROOT"] = self._get_fallback_dir()
 
+    def _fastcgi_deprecation(self, settings):
+        if settings.APPLICATION_SERVER in settings.FASTCGI_TYPES:
+            self.ctx.err(
+                "WARNING: FastCGI support is deprecated and will be removed"
+                " in OMERO 5.2. Install gunicorn and update config.")
+
+    def _assert_config_argtype(self, argtype, settings):
+        mismatch = False
+        if (settings.APPLICATION_SERVER in settings.WSGI_TYPES and
+                "wsgi" not in argtype):
+            mismatch = True
+        if settings.APPLICATION_SERVER in ("development",):
+            mismatch = True
+        if (settings.APPLICATION_SERVER in settings.FASTCGI_TYPES and
+                argtype not in ("nginx", "nginx-development",
+                                "apache", "apache-fcgi")):
+            mismatch = True
+        if mismatch:
+            self.ctx.die(1, ("ERROR: configuration mismatch. "
+                             "omero.web.application_server=%s cannot be used "
+                             " with 'omero web config %s'"
+                             ".") % (settings.APPLICATION_SERVER, argtype))
+            self._fastcgi_deprecation(settings)  # to be removed in 5.2
```

Comment:
@joshmoore which error code I should exactly use is there any pattern for that number. I temporally used 1